src/components/common/TransactionDetailsCategoryInput.js

```diff
@@ -0,0 +1,126 @@
+// @flow
+import { bns } from 'biggystring'
+import React, { Component } from 'react'
+import { View, TextInput, TouchableWithoutFeedback } from 'react-native'
+import s from '../../locales/strings.js'
+import { type AirshipBridge, AirshipModal } from '../modals/modalParts'
+import { sprintf } from 'sprintf-js'
+
+import { FormField } from '../common/FormField'
+import FormattedText from '../../modules/UI/components/FormattedText/index'
+import styles, { materialInput } from '../../styles/scenes/TransactionDetailsStyle'
+import THEME from '../../theme/variables/airbitz'
+import SubCategorySelect from '../common/TransactionSubCategorySelect.js'
+
+type Props = {
+  bridge: AirshipBridge<null>,
+  categories: Array<string>,
+  subCategories: Array<string>,
+  category: string,
+  subCategory: string,
+  onChange: (string, string) => void,
+  setNewSubcategory: (string, Array<string>) => void
+}
+
+type State = {
+  category: string,
+  subCategory: string
+}
+
+export class TransactionDetailsCategoryInput extends Component<Props, State> {
+  constructor (props: Props) {
+    super(props)
+    const { category, subCategory } = props
+    this.state = { category, subCategory }
+  }
+  onChangeCategory = (category: string) => {
+    this.setState({ category })
+    this.props.onChange(category, this.state.subCategory)
+  }
+  onChangeSubCategory = (subCategory: string) => {
+    this.setState({ subCategory })
+    this.props.onChange(this.state.category, subCategory)
+  }
+  onSelectSubCategory = (input: string) => {
+    const { bridge, subCategories, setNewSubcategory } = this.props
+    const splittedSubCategory = input.split(':')
+    const category = splittedSubCategory[0].toLowerCase()
+    const subCategory = splittedSubCategory[1]
```

Comment:
Subcategory is everything after the first : 
More colons may exist but need to be in the subcategory